zblue: hfp: ag: fix cannot connect ag after disconnect ag.

expliyh opened the pull request:
bug: v/81752

Rootcause: disconnected_callback not called caused connect info not cleared in connection manager module

zhongzhijie1 commented on service/stacks/zephyr/sal_hfp_ag_interface.c:
```suggestion
bt_sal_cm_profile_disconnected_callback(&bd_addr, PROFILE_HFP_AG, CONN_ID_DEFAULT);
hfp_ag_on_connection_state_changed(&bd_addr, PROFILE_STATE_DISCONNECTED, 0, 0);
```

expliyh replied:
Done

expliyh committed:
zblue: hfp: ag: fix cannot connect ag after disconnect ag.

bug: v/81968

Rootcause: disconnected_callback not called caused connect info not cleared in connection manager module

Signed-off-by: YuhengLi <liyuheng@xiaomi.com>
